### PR TITLE
Add Table of Contents to README

### PR DESCRIPTION
**Pull request by olwend**
#### Are there any manual testing steps?

---

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,25 @@
-# Speakerline
+<!-- # Speakerline -->
```
Comment (nodunayo) [inline README.md]: Can we keep the 'Speakerline' title?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,25 @@
...
 
 [![Build Status](https://travis-ci.org/nodunayo/speakerline.svg?branch=master)](https://travis-ci.org/nodunayo/speakerline)
 [![Dependency Status](https://gemnasium.com/badges/github.com/nodunayo/speakerline.svg)](https://gemnasium.com/github.com/nodunayo/speakerline)
 
 > Showcasing speakers' proposals and timelines in a effort to demystify the CFP process and help new speakers get started
 
-## Where did the idea come from?
+# Contents
```
Comment (nodunayo) [inline README.md]: Can we make this the same heading level as the headings, i.e. two hashes?

File: README.md [hunk 3]
```diff
@@ -1,11 +1,25 @@
...
+
+* [Where did the idea come from?](#where-did-the-idea-come-from)
+* [Goals of Speakerline](#goals-of-speakerline)
+* [Using the speakerline website](#using-the-speakerline-website)
```
Comment (nodunayo) [inline README.md]: Can we capitalise 'speakerline' here?

File: README.md [hunk 4]
```diff
@@ -1,11 +1,25 @@
...
+* [Setting up for local development](#setting-up-for-local-development)
+  * [Getting started with Ruby and Postgres](#get-started-Ruby-Postgres)
+  * [Getting the Rails app running](#getting-rails-app-running)
+    * [Environment variables](#environment-variables)
+  * [Running the tests](#running-the-tests)
+* [Contributing to Speakerline](#contributing-to-speakerline)
```
Comment (nodunayo) [inline README.md]: "Contributing to Speakerline" is in the same section as "Setting up for local development", so this needs indenting in by one.

File: README.md [hunk 5]
```diff
@@ -1,11 +1,25 @@
...
+* [Credits](#Credits)
+* [Copyright](#Copyright)
+
+## Where did the idea come from? <a name="where-did-the-idea-come-from"></a>
```
Comment (nodunayo) [inline README.md]: You've added the <a> tags to each heading but we don't need them. GitHub automatically moves you down to where you need to be on the page! :-)

**Comment by olwend** [comment]
Hi Nadia

Yes of course. To be honest I was rushed at end of session to submit. Happy
to rebase. I will drop the DB changes.

Thanks

Olwen


On 2 Dec 2017 5:04 pm, "Nadia Odunayo" <notifications@github.com> wrote:

*@nodunayo* requested changes on this pull request.

Hi @olwend <https://github.com/olwend>,

Thanks so much for tackling this!

Do you know how to do a git rebase?

If so, can you make this PR into two commits:

   1. The first separates out the 'db/schema' changes that have come about
   from me being on a later version of Rails
   2. A second one that has all the README changes and is entitled: "Add
   Table of Contents to README", as you've labeled this PR.

Let me know if any of my comments don't make sense or you need any help! :-)
------------------------------

In README.md
<https://github.com/nodunayo/speakerline/pull/42#discussion_r154500827>:

> @@ -1,11 +1,25 @@
-# Speakerline
+

Can we keep the 'Speakerline' title?
------------------------------

In README.md
<https://github.com/nodunayo/speakerline/pull/42#discussion_r154500838>:

>
 [Build Status](https://travis-ci.org/nodunayo/speakerline)
 [Dependency Status](https://gemnasium.com/github.com/nodunayo/speakerline)

 > Showcasing speakers' proposals and timelines in a effort to
demystify the CFP process and help new speakers get started

-## Where did the idea come from?
+# Contents

Can we make this the same heading level as the headings, i.e. two hashes?
------------------------------

In README.md
<https://github.com/nodunayo/speakerline/pull/42#discussion_r154500853>:

>
 [Build Status](https://travis-ci.org/nodunayo/speakerline)
 [Dependency Status](https://gemnasium.com/github.com/nodunayo/speakerline)

 > Showcasing speakers' proposals and timelines in a effort to
demystify the CFP process and help new speakers get started

-## Where did the idea come from?
+# Contents
+
+* [Where did the idea come from?](#where-did-the-idea-come-from)
+* [Goals of Speakerline](#goals-of-speakerline)
+* [Using the speakerline website](#using-the-speakerline-website)

Can we capitalise 'speakerline' here?
------------------------------

In README.md
<https://github.com/nodunayo/speakerline/pull/42#discussion_r154500894>:

>
 [Build Status](https://travis-ci.org/nodunayo/speakerline)
 [Dependency Status](https://gemnasium.com/github.com/nodunayo/speakerline)

 > Showcasing speakers' proposals and timelines in a effort to
demystify the CFP process and help new speakers get started

-## Where did the idea come from?
+# Contents
+
+* [Where did the idea come from?](#where-did-the-idea-come-from)
+* [Goals of Speakerline](#goals-of-speakerline)
+* [Using the speakerline website](#using-the-speakerline-website)
+* [Setting up for local development](#setting-up-for-local-development)
+  * [Getting started with Ruby and Postgres](#get-started-Ruby-Postgres)
+  * [Getting the Rails app running](#getting-rails-app-running)
+    * [Environment variables](#environment-variables)
+  * [Running the tests](#running-the-tests)
+* [Contributing to Speakerline](#contributing-to-speakerline)

"Contributing to Speakerline" is in the same section as "Setting up for
local development", so this needs indenting in by one.
------------------------------

In README.md
<https://github.com/nodunayo/speakerline/pull/42#discussion_r154500906>:

> -## Where did the idea come from?
+# Contents
+
+* [Where did the idea come from?](#where-did-the-idea-come-from)
+* [Goals of Speakerline](#goals-of-speakerline)
+* [Using the speakerline website](#using-the-speakerline-website)
+* [Setting up for local development](#setting-up-for-local-development)
+  * [Getting started with Ruby and Postgres](#get-started-Ruby-Postgres)
+  * [Getting the Rails app running](#getting-rails-app-running)
+    * [Environment variables](#environment-variables)
+  * [Running the tests](#running-the-tests)
+* [Contributing to Speakerline](#contributing-to-speakerline)
+* [Credits](#Credits)
+* [Copyright](#Copyright)
+
+## Where did the idea come from? <a name="where-did-the-idea-come-from"></a>

You've added the tags to each heading but we don't need them. GitHub
automatically moves you down to where you need to be on the page! :-)

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/nodunayo/speakerline/pull/42#pullrequestreview-80679340>,
or mute the thread

**Comment by nodunayo** [comment]
Hi Olwen,

Fine for you to keep the DB changes in — since I've seen that it's an inevitable change introduced in Rails 5.1.1 — just want those in a separate commit (within the same PR).

Did you notice the other changes I suggested too?

Thanks a lot!

Nadia

**Comment by olwend** [comment]
Really appreciate your comments. I will look tomorrow, I've just been
chatting to another lady who is working on another Speakerline PR. She had
issues with DB changes, which she resolved by setting a Postgres version.
I'm hoping to pair with her remotely tomorrow so thanks very much for
putting the PRs out there.

I'll be back in touch tomorrow PM.

Olwen

**Comment by nodunayo** [comment]
Hi Olwen,

This is fantastic. Thanks for letting me know. Were you at the 24PR Codebar event today by any chance?

The other lady's name doesn't happen to be Julie, does it? (Just noticed that she forked the repo at the same time you did. ☺️)

A good thing to do when you pick up a PR is to comment in the Issue so that nobody else comes along and starts working on it too. Can you tell your friend to comment on the one she's working on? You might have noticed that I tagged you in the 'Table of Contents' one.

Speak to you soon,

Nadia

**Comment by olwend** [comment]
Hi Nadia, 
Appreciate your help (my first 'real life' PR) and it is messy. So now Ive made the corrections and just need to rebase to separate DB schema commit for rails 5.1.1 and Readme.
I plan to:
rename first commit - DBschema change rails 5.1.1 
squash next 4 commits (corrections)
rename last commit - Readme.md changes

I am not sure if it is possible to remove the readme.md from first commit? 
Now I know what Im doing better (thankyou for being patient) I'm happy to redo the changes on a new branch and delete this PR -open a new one. It would be cleaner and separate DB from readme. 

Which would you prefer? 
Thanks

Olwen

**Comment by nodunayo** [comment]
Hi Olwen,

Thank you so much. I'm glad you're having a good experience for your first PR!

A couple of the links are broken. The text in the link has to be exactly the same as the heading. Can you check those out?

I think the two that need fixing are "Getting the Rails app running" and "Getting started with Ruby and Postgres". Probably a good idea to check the others then!

Once those changes are made, if you're able to rebase/squash the current PR, then do that! (Maybe you might need to force push...not sure how it works if you've already pushed the commits up...)

If it doesn't work, feel free to close this PR and open a new one.

Thank you for the feedback on me being patient. I always aim to be so that made me smile! :-)

Speak soon,

Nadia

**Comment by olwend** [comment]
Rework submitted without DBSchema in #46
Can resubmit db/schema.rb via fresh set up on Rails 5.1.2  postgres (PostgreSQL) 9.6.4 environment with another PR, with any changes user needs to make listed.

**Pull request closed without merging**